File: packages/jsii-rosetta/test/translations/calls/will_type_deep_structs_directly_if_type_info_is_available.cs
class BaseDeeperStruct
{
    public int A { get; set; }
}

class DeeperStruct : BaseDeeperStruct
{
    public int B { get; set; }
}

class OuterStruct
{
    public int Foo { get; set; }
    public DeeperStruct Deeper { get; set; }
}

public void Foo(int x, OuterStruct outer)
{
}

Foo(25, new OuterStruct { Foo = 3, Deeper = new DeeperStruct {
    A = 1,
    B = 2
} });

File: packages/jsii-rosetta/test/translations/calls/will_type_deep_structs_directly_if_type_info_is_available.java
public class BaseDeeperStruct {
    private Number a;
    public Number getA() {
        return this.a;
    }
    public BaseDeeperStruct a(Number a) {
        this.a = a;
        return this;
    }
}

public class DeeperStruct extends BaseDeeperStruct {
    private Number b;
    public Number getB() {
        return this.b;
    }
    public DeeperStruct b(Number b) {
        this.b = b;
        return this;
    }
}

public class OuterStruct {
    private Number foo;
    public Number getFoo() {
        return this.foo;
    }
    public OuterStruct foo(Number foo) {
        this.foo = foo;
        return this;
    }
    private DeeperStruct deeper;
    public DeeperStruct getDeeper() {
        return this.deeper;
    }
    public OuterStruct deeper(DeeperStruct deeper) {
        this.deeper = deeper;
        return this;
    }
}

public void foo(Number x, OuterStruct outer) {
}

foo(25, OuterStruct.builder().foo(3).deeper(DeeperStruct.builder()
        .a(1)
        .b(2)
        .build()).build());
